### PR TITLE
docs: add prometheus-operator example for jobset

### PR DESCRIPTION
**Pull request by googs1025**
Fix: https://github.com/kubernetes-sigs/jobset/issues/626

### CODE REVIEW

File: examples/prometheus-operator/prometheus.yaml [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+apiVersion: v1
+kind: Service
+metadata:
+  name: jobset-metrics
+  namespace: jobset-system
+spec:
+  type: NodePort
+  # Port mapping: Note if deploying with kind for testing, if docker mappings are not exposed when the cluster starts,
+  # kube port-forward needs to be used for port forwarding
+  # kubectl port-forward services/jobset-metrics  39090:9090 --address 0.0.0.0 -njobset-system
```
Comment (danielvegamyhre) [inline examples/prometheus-operator/prometheus.yaml:30]:
```suggestion
  # kubectl port-forward services/jobset-metrics  39090:9090 --address 0.0.0.0 -n jobset-system
```

File: examples/prometheus-operator/prometheus.yaml [hunk 2]
```diff
@@ -0,0 +1,78 @@
+# more details can be found
+# https://github.com/prometheus-operator/prometheus-operator/blob/main/Documentation/api.md#monitoring.coreos.com/v1.Prometheus
+apiVersion: monitoring.coreos.com/v1
+kind: Prometheus
+metadata:
+  name: jobset-metrics
+  namespace: jobset-system
+spec:
+  serviceAccountName: prometheus-jobset
```
Comment (danielvegamyhre) [inline examples/prometheus-operator/prometheus.yaml:9]: Does this service account need to be created before we apply the Prometheus CR? I see you create it lower in this file - have you tested this manually?

Comment (googs1025) [reply examples/prometheus-operator/prometheus.yaml:9]: I split into rbac.yaml

File: examples/prometheus-operator/prometheus.yaml (hunk 1 above)
Comment (googs1025) [reply examples/prometheus-operator/prometheus.yaml]: done

**Comment by danielvegamyhre** [comment]
@googs1025 Can you add a short readme in this folder explaining the steps to demo this example?

**Comment by googs1025** [comment]
> @googs1025 Can you add a short readme in this folder explaining the steps to demo this example?
done

https://github.com/googs1025/jobset/tree/docs_prometheus/examples/prometheus-operator

File: examples/prometheus-operator/prometheus.yaml (hunk 2 above)
Comment (danielvegamyhre) [reply examples/prometheus-operator/prometheus.yaml:9]: Why not just put the service account and RBACs above the Prometheus CR in the same file?

Comment (googs1025) [reply examples/prometheus-operator/prometheus.yaml]: done

File: examples/prometheus-operator/README.md [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+prometheus.monitoring.coreos.com/jobset-metrics created
+service/jobset-metrics created
+```
+
+```bash
+root@VM-0-5-ubuntu:/home/ubuntu# kubectl get pods -n jobset-system
+NAME                                         READY   STATUS    RESTARTS   AGE
+jobset-controller-manager-76767b599b-v8wcc   2/2     Running   0          6d22h
+prometheus-jobset-metrics-0                  2/2     Running   0          17s
+root@VM-0-5-ubuntu:/home/ubuntu# kubectl get svc -njobset-system
```
Comment (danielvegamyhre) [inline examples/prometheus-operator/README.md:48]: Last nit: can you add a space between "-n" and "jobset-system" please? it is more readable

Comment (googs1025) [reply examples/prometheus-operator/README.md]: sorry. fixed.

**Comment by danielvegamyhre** [comment]
/lgtm
/approve
Thanks!

**Pull request merged**